src/App/Osu/OsuDifficultyCalculator.cpp

```diff
@@ -399,33 +398,22 @@ double OsuDifficultyCalculator::calculateStarDiffForHitObjectsInt(std::vector<Di
 	// ****************************************************************************************************************************************** //
 
 	// initialize dobjects
+	const size_t numDiffObjects = (upToObjectIndex < 0) ? sortedHitObjects.size() : upToObjectIndex + 1;
 	const bool isUsingCachedDiffObjects = (cachedDiffObjects.size() > 0);
-	diffObjects.clear();
-	diffObjects.reserve((upToObjectIndex < 0) ? sortedHitObjects.size() : upToObjectIndex+1);
+	DiffObject* diffObjects;
```

Comment:
Pointer *star always next to the *name and not the type please (subjective style)